### ISSUE TITLE
Not possible to override "multi-line-text.html" in email templates

### ISSUE THREAD

**Issue by handplant** [verbb/formie#1982]
### Describe the bug

I want to override a few fields in an email notification. I followed the documentation and successfully overridden the "file-upload" field. I did the same for the 'multi-line-text' field, but unfortunately, the template is not being used. It always shows random text.

![Bildschirmfoto 2024-07-19 um 10 04 56](https://github.com/user-attachments/assets/efd1af2c-84d6-4397-b823-e975ebb6f149)
![Bildschirmfoto 2024-07-19 um 10 10 17](https://github.com/user-attachments/assets/98ff376e-dc51-452c-93e2-e72720100132)

### Steps to reproduce

I followed the documentation and created custom email templates. [https://verbb.io/craft-plugins/formie/docs/template-guides/email-templates](https://verbb.io/craft-plugins/formie/docs/template-guides/email-templates)

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

Craft Pro 4.10.6

### Plugin version

2.1.21

### Multi-site?

Yes

**Comment by engram-design** [comment]
So a few things here, firstly the email templates for fields will only be used for bulk variables like "All Form Fields". If you're selecting specific fields to output their content, that's different, and will use the value of the field.

That is to say, if you have something like this:

<img width="318" alt="image" src="https://github.com/user-attachments/assets/8e229ceb-e413-4006-aa1a-f03de09d2b96">

It's going to show the notification as:

<img width="945" alt="image" src="https://github.com/user-attachments/assets/33a1ce3d-9439-4a86-aa27-a7b066ab6879">

Note that the former shows things in a slightly more formatted way, using the email template for the field. You've got total control over that. The latter just shows the raw value of the field. Some fields actually still do use the email template when outputting just the field, because they are too complex to output as a single string (Repeater, Table, Element-fields, etc). Just wanted to clarify that first.

Secondly, the email preview is going to be a little different to a real email, and it's going to generate a fake value for the field.

It looks like from your screenshot that you're outputting just singular fields on their own?

**Comment by handplant** [comment]
I want to format a specific email for a service that extracts data from emails programmatically. For this, I need to format multiline text fields as follows: `{{ value|replace({'\r\n': '°', '\r': '°', '\n': '°'}) }}`.

I have started to override the fields. The `file-upload.html` template works, but the `multi-line-text.html` template does not.

file-upload.html
```
{% set elements = value.all() %}
{% if elements | length %}
{{ elements[0].filename }}
{% endif %}
```

multi-line-text.html
```
{{ value|replace({'\r\n': '°', '\r': '°', '\n': '°'}) }}
```

Notification config:
![Bildschirmfoto 2024-07-22 um 11 26 41](https://github.com/user-attachments/assets/379687da-82ea-4569-bfe9-d86b4ee73061)

Received Email
![Bildschirmfoto 2024-07-22 um 11 28 04](https://github.com/user-attachments/assets/e627538d-1da0-4a88-852c-801c0e8fd439)

----

Unfortunately, I don't really understand your response. It is definitely the case that overriding `file-upload` works, but `multi-line-text` does not.

What options do I have?

**Comment by engram-design** [comment]
As I mentioned, some fields like File Upload fields use their [email template](https://github.com/verbb/formie/blob/8ca2eef92548e1754534e8f3f939d9ecb98ea573/src/helpers/Variables.php#L488-L502) for the value in email notifications when referencing a single field like you're doing.

Other fields like Multi-Line Text call [getValueAsString](https://github.com/verbb/formie/blob/8ca2eef92548e1754534e8f3f939d9ecb98ea573/src/helpers/Variables.php#L426) and bypass the email template altogether.

That's why when you override the File Upload email template, it's working, but not for other fields like Multi-Line Text. It's only when you are using the consolidated variable tags like "All Form Fields" that always use the email template for fields. If you test using the "All Form Fields" variable, you'll see that the Multi-Line Text field uses your custom template.

It's done this way mostly to handle the two ways in which fields are output, on their own and part of their consolidated collection. Where you want just the value with no formatting for the former, and where you want formatting for the latter (for example, most email templates wrap content in a `<p>` and show the [label](https://github.com/verbb/formie/blob/craft-4/src/templates/_special/email-template/fields/multi-line-text.html)). But there are exceptions to this, like the File Upload field, whose value doesn't really translate to a single value and is rendered as a `<ul>`.

Changing this now would be a breaking change, but we'll improve this in Formie 3.

For right now I'm going to have to introduce a config setting for you to enforce the value of fields to use their email-based HTML. Add a [configuration](https://verbb.io/craft-plugins/formie/docs/get-started/configuration) setting to your `/config/formie.php` file to get the behaviour that you'd expect by setting `useEmailTemplateForFieldVariables` to `true`.
 
Updated for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.1.21"`.

**Comment by handplant** [comment]
Ok, thanks. I think I understand it now. :)

**Comment by engram-design** [comment]
Fixed in [2.1.23](https://github.com/verbb/formie/releases/tag/2.1.23)
